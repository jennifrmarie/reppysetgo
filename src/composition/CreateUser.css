.login__form__credentials {
    z-index: 1;
    position: relative;
    padding: 50px;
    min-height: 400px;
    display: flex;
    flex-direction: column;
    align-items: center;

}

.login__form__credentials:before {
    content: "";
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    width: 100%;
    height: 100%;
    background: url('https://images.pexels.com/photos/669576/pexels-photo-669576.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260');
    opacity: .5;
    background-size: cover;
    min-height: 800px;
    z-index: -1;
}

.login__header {
    font-family: 'Open Sans Condensed', sans-serif;
    color: white;
    /* letter-spacing: .5em; */
    font-size: 40px;
    text-align: center;
}

.username, .password {
    margin-bottom: 20px;
    font-size: 20px;
    color: black;
}
.logo__login {
    background-image: url('https://i.imgur.com/ZTJ52z4.png');
    background-size: cover;

    background-color: #ece8e8fb;
    background-position: center;
    font-size: 20px;
    position: relative;
    min-height: 400px;
    max-width: 100%;
    background-repeat: no-repeat;
}

.account_button {
    font-family: 'Open Sans Condensed', sans-serif;
    font-size: 30px;
    color: black;
    background-color: white;
    padding: 5px;
}

.account_button:hover {
    cursor: pointer;
}