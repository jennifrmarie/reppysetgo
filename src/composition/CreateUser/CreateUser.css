.login__form__credentials {
    z-index: 1;
    position: relative;
    padding: 50px;
    display: flex;
    flex-direction: column;
    align-items: center;
    height: calc(100vw * (9/16));
  max-height: calc(60em * (9/16));

}

.password_req {
    font-size: 10px;
}

.login__form__credentials:before {
    content: "";
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    width: 100%;
    background: url('https://images.pexels.com/photos/669576/pexels-photo-669576.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260');
    opacity: .5;
    background-size: cover;
    z-index: -1;
}

.login__header {
    font-family: 'Open Sans Condensed', sans-serif;
    color: black;
    font-size: 40px;
    text-align: center;
}

.username, .password {
    margin-bottom: 20px;
    font-size: 20px;
    color: black;
}
.logo__login {
    background-image: url('https://i.imgur.com/ZTJ52z4.png');
    background-size: 70%;

    background-color: #ece8e8fb;
    background-position: center;
    font-size: 20px;
    position: relative;
    height: 350px;
    max-width: 100%;
    background-repeat: no-repeat;
}

.account_button {
    font-family: 'Open Sans Condensed', sans-serif;
    font-size: 30px;
    padding: 5px;
}

.account_button:hover {
    cursor: pointer;
}

@media only screen and (max-width: 900px) {
    .logo__login {
        background-size: 100%;
        height: 200px;
    }

    .login__header {
        font-size: 25px;
    }

    .login__form__credentials {
        height: 100vh;
    }
}