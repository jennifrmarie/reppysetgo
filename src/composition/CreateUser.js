import React, { Component } from 'react'
import './CreateUser.css'
import config from '../config'

export default class CreateUser extends Component {

    onSubmit = (e) => {
        e.preventDefault()
        const user_name = e.target.user_name.value
        const password = e.target.password.value
        const data = { user_name, password }
        fetch(`${config.API_ENDPOINT}api/auth/login`, {
            mode: 'no-cors',
            method: 'post',
            headers: {
                "content-type": "application/json"
            },
            body: JSON.stringify(data)
        })
        .then(res => res.json())
        .then(data => {
            console.log(data)
            this.props.history.push('/')
        })
    }


    render() {
        return (
            
            <form onSubmit={this.onSubmit}>
                <div className="logo__login"></div>
                <div className="login__form__credentials">
                    <h3 className="login__header">Let's get started!</h3>
                    <input 
                        className="username" 
                        type="text" 
                        placeholder="Username" 
                         name="user_name" 
                         id="user_name" 
                    />
                    <input 
                        className="password" 
                        type="password" 
                        placeholder="Password"
                        name="password" 
                        id="password" 
                    />
                    <input 
                        className="password" 
                        type="password" 
                        placeholder="Re-enter Password"
                        name="passwordconfirm" 
                        id="passwordconfirm" 
                    />
                    <button 
                        className="account_button"
                    >
                        Create Account
                    </button>
                </div>
            </form>
        )
    }
}
