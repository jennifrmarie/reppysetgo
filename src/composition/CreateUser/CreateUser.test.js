import React from 'react';
import ReactDOM from 'react-dom';
import CreateUser from './CreateUser';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CreateUser />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CreateUser name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });