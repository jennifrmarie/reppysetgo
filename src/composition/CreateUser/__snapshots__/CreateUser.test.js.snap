// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="logo__login"
  />
  <div
    className="login__form__credentials"
  >
    <h3
      className="login__header"
    >
      Let's get started!
    </h3>
    <input
      className="username"
      id="user_name"
      name="user_name"
      placeholder="Username"
      required={true}
      type="text"
    />
    <input
      className="password"
      id="password"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <input
      className="password"
      id="passwordconfirm"
      name="passwordconfirm"
      placeholder="Re-enter Password"
      required={true}
      type="password"
    />
    <button
      className="account_button"
    >
      Create Account
    </button>
  </div>
</form>
`;
